src/shared/systems/serialization.hpp

```diff
@@ -0,0 +1,140 @@
+/*
+** EPITECH PROJECT, 2022
+** Manager.hpp
+** File description:
+** .
+*/
+
+#pragma once
+
+#include "../ECS/Components.hpp"
+#include "../ECS/ECS.hpp"
+#include "../ECS/Manager.hpp"
+#include <iostream>
+#include <regex>
+#include <sstream>
+#include <string>
+
+std::string toString(Armor component) {
+    std::stringstream ss;
+
+    ss << "(" << component.armor << ")";
+    return ss.str();
+}
+
+std::string toString(Health component) {
+    std::stringstream ss;
+
+    ss << "(" << component.health << ")";
+    return ss.str();
+}
+std::string toString(Position component) {
+    std::stringstream ss;
+
+    ss << "(" << component.xpos << "," << component.ypos << ")";
+    return ss.str();
+}
+
+template <class T>
+std::string componentToString(EntityID entityId, Manager* manager) {
+    T* component = manager->getComponent<T>(entityId);
+
+    return std::to_string(getID<T>()) + toString(*component);
+}
+
+template <class... ComponentTypes>
+std::string entityToString(EntityID entityID, Manager* manager) {
+    Index entityIndex = getIndex(entityID);
+
+    std::stringstream stream;
+
+    stream << entityIndex << "(";
+
+    bool first = true;
+    for (const std::string component : {componentToString<ComponentTypes>(entityID, manager)...}) {
+        if (!first)
+            stream << ";";
+        else
+            first = false;
+
+        stream << component;
+    }
+
+    stream << ")";
+
+    return stream.str();
+}
+
+std::vector<std::string> argsSplit(const std::string component) {
+    std::regex argsRegex("\\(.*\\)", std::regex_constants::ECMAScript);
+    std::smatch match;
+
+    std::regex_search(component.begin(), component.end(), match, argsRegex);
+    std::vector<std::string> args;
+    std::regex argRegex("\\d+(\\.\\d+)?", std::regex_constants::ECMAScript);
+
+    std::sregex_iterator words_begin(match.str().begin(), match.str().end(), argRegex);
+    std::sregex_iterator words_end = std::sregex_iterator();
+
+    for (std::sregex_iterator i = words_begin; i != words_end; ++i) {
+        args.push_back((*i).str());
+    }
+
+    return args;
+}
```

Comment:
This seems way more complicated than needed. Why do we even need to have the parentheses ?
Without parentheses we would still be able to serialize entities by using two different separators which would make the code wayyyy more simple.

I strongly feel this is overcomplicated.